### DISCUSSION TITLE
15.7.0 Setup Wizard cannot take a Reference Picture

### DISCUSSION THREAD

**Discussion by eddyjay** [jomjol/AI-on-the-edge-device#3640] [General]
### The Problem

On a brand new install with 15.7.0 During setup process, live view for focus works fine. As soon as in the next page the button "create new reference" is clicked, the system will never return from "waiting for picture". Icon showing javascript issues is also triggered. 
Reverted back to 15.6.0, and no problems

### Version

15.7.0

### Logfile

```shell
Cannot access logviewer
```

**Comment by claasgo** [comment]
I have the same problem. I deleted the reference images manually and then it worked. But in general 15.7.0 runs very unstable and slow for me.

**Comment by arkoode** [comment]
Same problem, downgraded too 15.6.0 and everything works fine ...

**Comment by mattg2** [comment]
not very handy, but starting over from scratch is working

**Comment by Pewibe** [comment]
Because previous comments have no log I like to add my log. It shows 2 times the situation where I clicked the "create new reference" button (please search for "backtrace").
[2024-02-28 Show bug in v15.7.0.txt](https://github.com/jomjol/AI-on-the-edge-device/files/14439004/2024-02-28.Show.bug.in.v15.7.0.txt)
The RSSI is excellent because there is 30 cm distance :-)

**Comment by stidch** [comment]
It neither works in my case in
`Settings` - `Alignemnt` - `Reference Image`

**Comment by mattg2** [comment]
I started over with fresh install and it works. I guess it has something to do with the new options in the config file, some of them looked wired in the web dialogue.

**Comment by Pewibe** [comment]
What is your definition of fresh install? Just install v15.7.0 again in ESP32 or do even other steps? What about SD card - just leave it?

**Comment by SybexX** [comment]
https://github.com/jomjol/AI-on-the-edge-device/discussions/2926

**Comment by Pewibe** [comment]
A short answer might be efficiant. I searched the large #2926 for the words "fresh" (0 hits) and "install" (2 hits) but got no idea what steps are necessary for a "fresh install".

**Comment by mattg2** [comment]
Just start over like install the 1st time. That worked for me.

**Comment by ckoeber83** [comment]
Same for me Release: v15.7.0 results in reboot by taking Reference Picture and picture was quite dark in my case. Number recognition works not well. Revert back to Release: v15.6.0 (Commit: 8d6f731+) and problems are gone.

Stay tuned...

**Comment by enjoy-development** [comment]
same with Release v15.7.0 . 
Message : Taking new Image 
after a while it results in reboot by taking Reference Picture 
i also tried to use a new SD Card , deleted old Reference Picture , etc.

**Comment by enjoy-development** [comment]
i reset my SD Card Content by Restore from Backup. Reading the values works again. but still no new Reference image is possible

**Comment by drose030** [comment]
Same here, tried to reinstall from the start, nothing helped and now reverted to 15.6.0.

**Comment by yozik04** [comment]
I am also not able to create a new reference image, it crashes and goes into reboot. 15.7.0.
```
[0d00h00m05s] 2024-05-31T11:19:11 <WRN> [MAIN] Reset reason: Exception/panic
[0d00h00m05s] 2024-05-31T11:19:11 <WRN> [MAIN] Device was rebooted due to a software exception! Log level is set to DEBUG until the next reboot. Flow init is delayed by 5 minutes to check the logs or do an OTA update
[0d00h00m05s] 2024-05-31T11:19:11 <WRN> [MAIN] Keep device running until crash occurs again and check logs after device is up again
```

Nothing interesting in DEBUG log level as well.

**Comment by TungstenE2** [comment]
same issue here...

Issue is pending for 4 months. Anybody looking into it?

![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/39955685/2fed5f98-c95e-4064-abfe-6acae2266a15)

**Comment by yozik04** [comment]
Release: v15.6.0 (Commit: 8d6f731+) works perfectly for me. I had to downgrade.

**Comment by gbossss** [comment]
Taking a new reference picture did not work with 15.7.0
A downgrade ("update") to 15.6.0 helped ...

**Comment by SybexX** [comment]
If you want, you can try this version https://github.com/jomjol/AI-on-the-edge-device/actions/runs/10028173125 to see if that solves the problem.

**Comment by StephanTexxpro** [comment]
Works for me.

**Comment by dominikandreas** [comment]
> If you want, you can try this version https://github.com/jomjol/AI-on-the-edge-device/actions/runs/10028173125 to see if that solves the problem.

having issues with this version as well. I can create reference images, but for some reason the images in this version are very bright, the exposure control does not have any effect. First taken image worked, afterwards brightness continued to increase for some reason. reverting to 15.6.0

**Comment by SybexX** [comment]
@dominikandreas Have you tried the latest rolling yet? https://github.com/jomjol/AI-on-the-edge-device/actions/runs/10690676061

**Comment by dominikandreas** [comment]
@SybexX just tried the newest rolling, same issue. Can't take a picture that is not way too bright no matter what settings I choose. Seems like auto-exposure is broken. Also tried all sorts of combinations in expert settings.
![Screenshot_20240908-084600](https://github.com/user-attachments/assets/5048feb7-cae8-41b3-8c2a-547a0da2175e)
Before, these settings would've produced a perfectly fine image (also tried without expert settings).

**Comment by Ingo-Knito** [comment]
Have you “touched” and “moved” every single element when setting the settings - i.e. changed them once? I also have problems with the settings and have noticed that (at least once) not all settings have been saved in the settings.ini.

How many entries does the [TakeImage] section contain?

**Comment by dominikandreas** [comment]
I've now tried all sorts of things, even complete reinstall from scratch using the latest rolling. At first I was able to configure everything from scratch and it seemed to work, but after a while it also produced these over-exposed images (even though I didn't change any settings). No matter what I tried, I couldn't get it to take a normally exposed image anymore. After rolling back to 15.6.0 everything seems to be working again.

**Comment by SybexX** [comment]
If you delete all camera settings from config.ini, the camera's default settings will be used.
However, you are not allowed to do this via the reference page and configuration page, either on the PC or via the config.ini editor.
![Unbenannt](https://github.com/user-attachments/assets/0ad6a6d6-ce0f-44b7-8e68-e6e9f7fdccf6)
If you have problems with overexposure, reduce CamGainceiling and turn off Auto-Exposure Control 2.
You can find a good description of the camera parameters on this page: https://heyrick.eu/blog/index.php?diary=20210418&keitai=0

**Comment by SybexX** [comment]
@dominikandreas If you change the camera settings on the reference page, you must click "Update reference image" to see the result.

**Comment by dominikandreas** [comment]
There's obviously some over exposure bug going on here. After having chosen a good setting it suddenly got into over exposure mode after a reboot and no setting change in the UI would fix it. Unfortunately I already spent all the time I've had to play around with this and will stay on 15.6.0

**Comment by SybexX** [comment]
@dominikandreas If you still feel like it, you can test this firmware: https://github.com/SybexX/AI-on-the-edge-device/actions/runs/10761151230
The standard camera settings were adjusted in rolling because many complained that the image was too dark.
You're the first to complain that the picture is too bright^^

**Comment by dominikandreas** [comment]
When others complained about low brightness, it was probably related to some parameters / default setting in the code. In my case it's definitely a bug. I can configure the settings (used grayscale, led at 20) and capture a perfectly nice image, but then suddenly the image goes way too bright, even though I'm not changing any settings. I did enable grayscale conversion because I found it makes the image sharper, increased contrast and changed the LED intensity. Not sure if this could have a casual relationship to the over exposure problem. All this after a factory reset and configuring everything from scratch through the UI

**Comment by SybexX** [comment]
@dominikandreas In version 15.6.0 the grayscale conversion was done by ESP, in 15.7.0 and rolling version by the camera itself, so I suspect your camera quality is not that good.

**Comment by dominikandreas** [comment]
I was able to take a perfectly fine grayscale image before suddenly getting the over exposure problem, which appeared without changing any settings and no setting change could fix it. The only remedy after the problem appeared each time was either a factory reset or downgrade. So no, it has nothing to do with my camera quality.

**Comment by SybexX** [comment]
If it works with version 15.6.0 and not with version 15.7.0 and the current rolling, it's definitely your camera's fault.
I tested the negative mode and the grayscale mode on my watermeter for several days and never had any problems!
As already said, everything is done by the camera, the ESP only sends the camera the user settings and receives the finished image.

**Comment by dominikandreas** [comment]
Hmm, not sure how this could be camera related since it does work with 15.7.0 initially and only enters the "over-exposure mode" after a while. It seems to me like the user settings are somehow not sent correctly to the camera. Maybe this only occurs with my specific combination of settings (I configured rotation, LED intensity, played with auto exposure and grayscale)

**Comment by SybexX** [comment]
Either the camera's software is faulty and it freezes at a certain combination of settings, or the camera gets very hot after a certain time because it now has to do more calculations. When I worked with @jasaw on implementing the OV5640, we discovered that with some combinations of settings, either my camera or his camera was causing problems.

**Comment by yozik04** [comment]
I had to switch to Auto Exposure Control 2 to beat over-exposure:
![image](https://github.com/user-attachments/assets/54503f9f-3ec9-442c-a90b-2ec3e39cd14f)

**Comment by jasaw** [comment]
Just want to share my experience playing with my 3 units of OV2640.

The OV2640 gets warm if you don't attach it to a heat sink. Normally the cameras are supposed to be attached on the ESP32Cam's microSD card slot using a thermal tape, effectively using the ESP32Cam's PCB as a heat sink. My OV2640 without heat sink worked for a while until it overheated but another identical setup with heat sink never have any issue. Like what @SybexX said, the camera may encounter stability issues if it has to work harder and it is running outside of it rated temperature specification.

**Comment by friedpa** [comment]
I can also confirm with my three systems, since attaching the OV2640 to PCB I had no more problems with overheating CAMs.
